### ISSUE TITLE
Spec augmentation time masking

### ISSUE THREAD

**Issue by kssteven418** [NVIDIA/NeMo#3298]
**Describe the bug**

I was wondering if this line of [code](https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/asr/parts/numba/spec_augment/spec_aug_numba.py#L286) for time masking of spec augmentation is correct.
Unless intended, it looks more reasonable with `time_starts = torch.cat(time_starts, 0)` rather than `time_starts = torch.cat(time_lengths, 0)` as this line of code makes `time_starts` and `time_lengths` equivalent to each other.

**Steps/Code to reproduce bug**

Any scripts useing spec augmentation, e.g., `python speech_to_text_bpe.py --config-path conf/conformer --config-name conformer_ctc_bpe`

**Expected behavior**

It currently generates time masks only at the beginning of the sequence.
![Screenshot from 2021-12-07 22-47-11](https://user-images.githubusercontent.com/50283958/145161693-8069705e-3599-494c-89b4-70eec6140545.png)

**Environment overview (please complete the following information)**

`pip install -e .`

**Environment details**

N/A

**Additional context**

N/A

**Comment by titu1994** [comment]
Oh boy. Good catch, that's a bug. I create the time_starts list above but assign both as time_lengths concat. I'll send a but fix tomorrow.

**Comment by titu1994** [comment]
What's up with the time mask from 750 all the way to 1600? Is the time mask that large ?

**Comment by kssteven418** [comment]
Oh no, that's padding tokens after batching. Only the front ones are the time masking from specaugmentation

**Comment by titu1994** [comment]
Sent a PR to fix this, thankfully it shouldn't affect too much since the numba kernel is activated only when numba version matches and with cuda. Thanks for finding this.
